src/NServiceBus.Transport.Msmq/MsmqAddress.cs

```diff
@@ -48,14 +49,37 @@ public static MsmqAddress Parse(string address)
             return new MsmqAddress(queue, machineName);
         }
 
-        public bool IsRemote => Machine != RuntimeEnvironment.MachineName;
+        public bool IsRemote => Machine != RuntimeEnvironment.MachineName && IsLocalIpAddress(Machine) != true;
```

Comment:
How time consuming is this? Should it be lazily evaluated?

Reply:
As discussed, the remote check is done only at startup and we do not have to worry about performance impact. I've modified it from property to function to imply that this isn't a cheap operation.